net-utils/src/lib.rs

```diff
@@ -441,9 +581,15 @@ fn udp_socket_with_config(config: SocketConfig) -> io::Result<Socket> {
 pub fn bind_common_in_range(
     ip_addr: IpAddr,
     range: PortRange,
+    config: Option<SocketConfig>,
```

Comment:
there are a few of these functions i changed to take in an option. Not sure that is the best call. We could:
1) leave it as is, taking in an Option
2) create a new function `bind_common_in_range_with_config` and pass in SocketConfig
3) modify the original function to just take a SocketConfig